simple_history/management/commands/clean_duplicate_history.py

```diff
@@ -65,11 +65,21 @@ def _process(self, to_process, date_back=None, dry_run=True):
             if not found:
                 continue
 
+            # Break apart the query so we can add additional filtering
+            model_query = model.objects.all()
+
+            # If we're provided a stop date take the initial hit of getting the
+            # filtered records to iterate over
+            if stop_date:
+                model_query = model_query.filter(
+                    pk__in=(m_qs.values_list(model._meta.pk.name).distinct())
+                )
+
             # it would be great if we could just iterate over the instances that
```

Comment:
does this change not differ from the comment here? Is this change actually faster?

Reply:
If so, should we remove this comment?   @craigmaloney

Reply:
The comment still stands if the user doesn't suggest a date to work from. If no date is specified then it'll still crawl through all of the records.

Reply:
But it looks like the comment mentions a given period, which would make me assume it's talking about the filtered qs

Reply:
I'm not sure what to do here then, because my understanding is that the comment still applies if no date is specified (eg: it doesn't just search on the changes). I left it alone because I wasn't certain if it still applied or not. My understanding is that it does.

Reply:
I believe the comment is describing exactly what this change is doing. Have you done any testing to see which is faster?

Reply:
I have tested it on my database, with my patch, which actually completed (the previous version didn't complete because it was stuck checking records from 2003).

Reply:
Cool. Then let's remove that comment, and then I'll merge it. Sound good?

Reply:
Removed the comment.